main.go

```diff
@@ -175,9 +182,33 @@ func main() {
 		os.Exit(1)
 	}
 
-	setupLog.Info("starting manager")
-	if err := mgr.Start(ctx); err != nil {
-		setupLog.Error(err, "problem running manager")
-		os.Exit(1)
+	if !dryRun {
+		setupLog.Info("starting manager")
+		if err := mgr.Start(ctx); err != nil {
+			setupLog.Error(err, "problem running manager")
+			os.Exit(1)
+		}
+	} else {
+
+		setupLog.Info("Starting Plan for controller class " + controllerClass)
+		controlPlanes := &controlplanev1alpha1.KopsControlPlaneList{}
+		err := mgr.GetAPIReader().List(ctx, controlPlanes)
```

Comment:
why using the client.Reader instead of the client.Client? What is the difference between then?